go.mod

```diff
@@ -2,51 +2,52 @@ module github.com/pulumi/pulumi-yaml
 
 go 1.19
 
+replace github.com/hashicorp/vault/api v1.8.2 => github.com/hashicorp/vault/api v1.9.0
```

Comment:
github.com/hashicorp/vault/api v1.8.2 doesn't compile correctly, v1.9.0 has the fixes.